### PR TITLE
add IsDeployEnabled push test

### LINKED ISSUE #123: Make deployment opt-in
so it won't naturally happen for repos we don't generate

### LINKED ISSUE #159: Use PushTests and DSL for build and deploy
Replace `ConditionalBuilder`

`whenPushSatisfies(Test1).buildWith(FooBuilder)`

`whenPushSatisfies(Test1).deployWith(CloudFoundryDeploer)`

### PR DESCRIPTION
**Pull request by cdupuis**
Opt in and ouf of deploy with `@atomist enable deploy` and `@atomist disable deploy`. This will store a fact in the Graph called `DeployEnablement` which is defined as custom ingester.

fixes #159, fixes #123

### CODE REVIEW

File: src/handlers/commands/ToggleDeployEnablement.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    @MappedParameter(MappedParameters.GitHubRepository)
+    public repo: string;
+
+    @MappedParameter(MappedParameters.GitHubRepositoryProvider)
+    public providerId: string;
+
+}
+
+function toggleEnablement(enable: boolean) {
```
Comment (jessitron) [inline src/handlers/commands/ToggleDeployEnablement.ts]: Can we call this setEnablement or saveEnablement? It is not a toggle
Which makes me happy, toggles are evil

File: src/atomist.config.ts [hunk 2]
```diff
@@ -48,6 +49,10 @@ export const configuration: Configuration = {
     teamIds: extractWorkspacesFromEnvironment(),
     commands: assembled.commandHandlers.concat([]),
     events: assembled.eventHandlers.concat([]),
+    // TODO CD move ingesters to different global automation
+    ingesters: [
```
Comment (jessitron) [inline src/atomist.config.ts]: Will this prevent people from running it? 
If so
I think we have to move it before merging

Comment (cdupuis) [reply src/atomist.config.ts]: It won’t for as long as we don’t deploy global and durable. I was planning to deal with this separately.

File: src/handlers/commands/ToggleDeployEnablement.ts (hunk 1 above)
Comment (cdupuis) [reply src/handlers/commands/ToggleDeployEnablement.ts]: For sure. I’ll rename as you like.

**Comment by jessitron** [comment]
I can’t figure out how to approve on my phone. GitHub makes approval hard to find we could fix that!

**Comment by cdupuis** [comment]
No worries. This can wait a couple hours. Thanks.

File: src/atomist.config.ts (hunk 2 above)
Comment (johnsonr) [reply src/atomist.config.ts]: This seems OK to me

**Pull request merged**